ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/PluginsReloadConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.plugin.BesuPlugin;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PluginsReloadConfiguration implements JsonRpcMethod {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Map<String, BesuPlugin> namedPlugins;
+
+  public PluginsReloadConfiguration(final Map<String, BesuPlugin> namedPlugins) {
+    this.namedPlugins = namedPlugins;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.PLUGINS_RELOAD_CONFIG.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    try {
+      final String pluginName = requestContext.getRequiredParameter(0, String.class);
+      if (!namedPlugins.containsKey(pluginName)) {
+        LOG.info("Cannot reload plugin because not registered.");
```

Comment:
Also, do we have guidelines on what we consider an error for RPC? For example, is providing the tight number of parameters but the parameter misspelled a success or error? It looks like here we consider any validly formed RPC a success, regardless of the intention of the RPC. Not saying we should change it, genuinely curious about what the convention is here.

Reply:
I don't know if we have such guidelines honestly.  Maybe worth raising this internally.

Reply:
If you look at `EthSendRawTransaction`, failure to decode the transaction is grounds for an RPC error, even though the request was properly formed

Reply:
The spec implies that an error is for any kind of error associated with the RPC:
https://www.jsonrpc.org/specification#response_object

Reply:
Do you consider we should throw an error if a plugin not registered is specified ?

Reply:
Basically this line:
`This member MUST NOT exist if there was no error triggered during invocation.`

And above in the request section:
`A String containing the name of the method to be invoked.`

They're using invoke for the ultimate method called (reload in this case) so I think we should treat logic errors as RPC errors as well

Reply:
After looking at the spec, yeah, I think we should error on an invalid plugin name.

Reply:
ok sounds good. done.